packages/h5wasm/src/h5wasm-api.ts

```diff
@@ -75,12 +70,24 @@ export class H5WasmApi extends ProviderApi {
   }
 
   private async initFile(buffer: ArrayBuffer): Promise<H5WasmFile> {
-    await ready;
+    await h5wasmReady;
+
+    // Write file to Emscripten virtual file system
+    // https://emscripten.org/docs/api_reference/Filesystem-API.html#FS.writeFile
+    FS?.writeFile(this.filepath, new Uint8Array(buffer), { flags: 'w+' });
 
-    FS.writeFile(this.filepath, new Uint8Array(buffer), { flags: 'w+' });
     return new H5WasmFile(this.filepath, 'r');
   }
 
+  private async getH5WasmEntity(path: string): Promise<H5WasmEntity> {
+    const file = await this.file;
+
+    const h5wEntity = file.get(path) || undefined;
+    assertDefined(h5wEntity, `No entity found at ${path}`);
```

Comment:
Note that `|| undefined` is due to the fact that `assertDefined` asserts only for `undefined`, not `null` (which is a good thing, we don't want to change that). I felt creating an `assertNonNull` guard was unnecessary.